client/extensions/woocommerce/store-sidebar/index.js

```diff
@@ -45,34 +46,31 @@ class StoreSidebar extends Component {
 		site: PropTypes.object,
 	};
 
-	componentDidMount = () => {
-		const { productsLoaded, site } = this.props;
+	componentDidMount() {
+		const { site } = this.props;
 
 		if ( site && site.ID ) {
-			this.fetchData( { siteId: site.ID, productsLoaded } );
+			this.fetchData();
 		}
-	};
+	}
 
-	componentWillReceiveProps = newProps => {
+	componentDidUpdate( prevProps ) {
 		const { site } = this.props;
```

Comment:
While we are here should we utilize `props.siteId` instead to simplify this logic?